feat(design-system): Integrate Inkline component library as baseline

tkaragianes opened the pull request:
## Summary
We plan to integrate the Inkline Vue component library as a starting point to speed component creation. We will wrap Inkline components in an Eterna components and use the Eterna components in our pages. This gives us an escape hatch should we need to replace the Inkline library in the future, and lets us define Eterna-specific details for each component.

## Implementation Notes
Inkline uses CSS variables to style components. Eterna uses an override file that allows us to globally reassign Inkline's variables with our design-system standards. We can also locally override Inkline variables on a per-component basis.

## Related Issues
Closes #40.

tkaragianes committed:
Integrate Inkline component library as baseline

We plan to integrate the Inkline Vue component library as a starting
point to speed component creation. We will wrap Inkline components in
an Eterna components and use the Eterna components in our pages. This
gives us an escape hatch should we need to replace the Inkline library
in the future, and lets us define Eterna-specific details for each
component.

tkaragianes committed:
Update package-lock.json

luxaritas commented on packages/design-system/vue/package.json:
```diff
@@ -15,10 +15,15 @@
+    "./style.css": "./src/assets/style.css"
   },
   "dependencies": {
+    "@inkline/inkline": "^3.2.0",
```
This should also have design-system-tokens as a dependency, right?

tkaragianes replied:
It should.

luxaritas reviewed (commented):
It doesn't look like this closes #40 - there's still a lot of Inkline global variables that aren't mapped to tokens

tkaragianes committed:
Add design-system-tokens dependency to design-system-vue

tkaragianes committed:
Update Inkline overrides

tkaragianes committed:
Update build:watch script to transform on initial run

Build:watch on the token-transformer now runs the tranform script
on initial call, and then watches tokens.json and retriggers the
transform script on changes.

tkaragianes committed:
Update token transformer to accomodate multi-property tokens

Token transformer now splits multi-property values into separate css
variables. This commit also removes the font-family hack now that
Figma supports comma separated values.

tkaragianes committed:
Update tokens with color themes: default, dark, light

tkaragianes committed:
Fix token transformer multi-property conversion

Our TS rules don't allow variable shadowing. In correcting the issue,
I missed one `key` that should be `property`.

tkaragianes committed:
Update inkline overrides
